compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -0,0 +1,108 @@
+package dotty.tools
+
+
+import scala.annotation.tailrec
+import scala.io.Source
+import scala.util.Try
+import java.net.URLClassLoader
+
+enum ExecuteMode:
+  case Guess
+  case Script
+  case Repl
+  case Run
+
+case class Settings(
+  verbose: Boolean = false,
+  classPath: List[String] = List.empty,
+  executeMode: ExecuteMode = ExecuteMode.Guess,
+  exitCode: Int = 0,
+  residualArgs: List[String] = List.empty,
+  scriptArgs: List[String] = List.empty,
+  targetScript: String = "",
+) {
+  def withExecuteMode(em: ExecuteMode): Settings = this.executeMode match
+    case ExecuteMode.Guess =>
+      this.copy(executeMode = em)
+    case _ =>
+      println(s"execute_mode==[$executeMode], attempted overwrite by [$em]")
+      this.copy(exitCode = 1)
+  end withExecuteMode
+
+  def withResidualArgs(args: String*): Settings =
+    this.copy(residualArgs = residualArgs.appendedAll(args.toList))
+
+  def withScriptArgs(args: String*): Settings =
+    this.copy(scriptArgs = scriptArgs.appendedAll(args.toList))
+
+  def withTargetScript(file: String): Settings =
+    Try(Source.fromFile(file)).toOption match
+      case Some(_) => this.copy(targetScript = file)
+      case None      =>
+        println(s"not found $file")
+        this.copy(exitCode = 2)
+  end withTargetScript
+}
+
+object MainGenericRunner {
+
+  @tailrec
+  def process(args: List[String], settings: Settings): Settings = args match
+    case Nil =>
+      settings
+    case "-repl" :: tail =>
```

Comment:
In which situation is it necessary to explicitly pass -repl?

Reply:
Actually, never. But `dist/bin` script provides such option, so we wanted to recreate this

Reply:
If it's never used let's just remove it then.

Reply:
> In which situation is it necessary to explicitly pass -repl?

In `dist/bin/scalac`, it's needed, but doesn't seem to be needed in `dist/bin/scala`.